src/tests/ilp-kit.js

```diff
@@ -2,37 +2,79 @@
 'use strict'
 
 const path = require('path')
-// const assert = require('assert')
 const ServiceManager = require('five-bells-service-manager')
 const KitManager = require('../lib/kit-manager')
-// const fs = require('fs')
 const request = require('superagent')
 const assert = require('assert')
+const spawn = require('child_process').spawn
 
 const services = new ServiceManager(
   path.resolve(process.cwd(), 'node_modules/'),
   path.resolve(process.cwd(), 'data/'))
 const kitManager = new KitManager(services)
 
 const configFiles = [ require.resolve('../tests/data/kit1-env.list'),
-                      require.resolve('../tests/data/kit2-env.list'),
-                      require.resolve('../tests/data/kit3-env.list')]
+                      require.resolve('../tests/data/kit2-env.list')]
+                      // if more complex test cases require more ilp kit instances,
+                      // add more env.list files below.
+                      // require.resolve('../tests/data/kit3-env.list')]
 
 // sleep time expects milliseconds
 function sleep (time) {
   return new Promise((resolve) => setTimeout(resolve, time))
 }
 
+function assertStatusCode (resp, expectedStatus) {
+  assert.equal(resp.statusCode, expectedStatus, `HTTP status code is ${resp.statusCode}, 
+        but expected is ${expectedStatus}`)
+}
+
+// sets up Apache as a reverse-proxy and for handling SSL connections
+function * setupApache () {
+  const image = process.env.CIRCLE_BUILD_IMAGE
+  if (!image) {
+    console.log('WARN: Not running on CircleCI. ' +
+      'Please setup Apache configuration manually.')
+  } else if (image !== 'ubuntu-14.04') {
+    throw new Error('Incompatible build image, use Ubuntu 14.04 instead.')
+  } else {
+    try {
+      const scriptPath = require.resolve('../../assets/ci/setup_ssl.sh')
+      const scriptDir = path.resolve(scriptPath, '..')
+      spawn('sh', [scriptPath], {
+        cwd: scriptDir,
+        stdio: 'inherit'
+      })
+      yield sleep(5000)
+    } catch (e) {
+      throw new Error('Failed to setup Apache as a reverse-proxy: ' + e.message)
+    }
+  }
+}
+
 describe('ILP Kit Test Suite -', function () {
   before(function * () {
     try {
+      // accept self-signed certificates
+      process.env['NODE_TLS_REJECT_UNAUTHORIZED'] = 0
+
+      yield setupApache()
+
       let startupPromises = []
       for (const config of configFiles) {
         const p = kitManager.startKit(config)
         startupPromises.push(p)
       }
       yield Promise.all(startupPromises)
-      yield sleep(10000) // let the ilp-kits start
+      yield sleep(50000) // let the ilp-kits start
```

Comment:
That's no good - timeouts are brittle and are going to cause random failures. You can wait for `ilp-kit` to print a key phrase on stdout (check out how the other components are started for inspiration) or you can wait on an HTTP endpoint becoming available. I'd recommend the former, because it doesn't rely on polling.

If there isn't a suitable log message in ilp-kit that you can listen for, feel free to add one.